src/Core/Services/Implementations/EventService.cs

```diff
@@ -185,6 +185,14 @@ private async Task CreateLogGroupEventAsync(Group group, EventType type, EventSy
             Date = date.GetValueOrDefault(DateTime.UtcNow),
             SystemUser = systemUser
         };
+
+        if (systemUser is EventSystemUser.SCIM)
```

Comment:
All Events with a defined `SystemUser` should have the new `DeviceType.Server` value so I'd move that logic up to the `EventMessage` constructor:

```
var e = new EventMessage(_currentContext)
        {
            OrganizationId = group.OrganizationId,
            GroupId = group.Id,
            Type = type,
            ActingUserId = _currentContext?.UserId,
            ProviderId = await GetProviderIdAsync(@group.OrganizationId),
            Date = date.GetValueOrDefault(DateTime.UtcNow),
            SystemUser = systemUser,
            DeviceType = systemUser.HasValue ? DeviceType.Server : null
        };
```

Reply:
That makes perfect sense, I'm happy with it being this way.